contracts/programs/bubblegum/src/lib.rs

```diff
@@ -346,6 +409,43 @@ pub mod bubblegum {
         gummyroll::cpi::init_empty_gummyroll(cpi_ctx, max_depth, max_buffer_size)
     }
 
+    pub fn transfer_tree_owner(
+        ctx: Context<TransferTreeOwner>,
+    ) -> Result<()> {
+        ctx.accounts.authority.owner = ctx.accounts.new_owner.key();
+        Ok(())
+    }
+
+    pub fn transfer_tree_delegate(
+        ctx: Context<TransferTreeDelegate>,
+    ) -> Result<()> {
+        ctx.accounts.authority.delegate = ctx.accounts.new_delegate.key();
+        Ok(())
+    }
+
+    pub fn add_append_authority(
+        ctx: Context<SetTreeAppendAuthority>,
+        num_appends: u64,
+    ) -> Result<()> {
+        let empty_key: [u8; 32] = [0; 32];
+        for (i, node) in ctx.accounts.authority.append_allowlist.iter().enumerate() {
```

Comment:
It is also prudent to check that this authority is not already in the `append_allowlist`